### DISCUSSION TITLE
Using Tethys as a CMS

### DISCUSSION THREAD

**Discussion by JamesWanjohi** [tethysplatform/tethys#669] [Ideas]
Apologies if this might be trivial.

I would like to know how I can use Tethys as a content management system, that is to help manage a home page content (images, text) that keeps changing every now and then as new information is created.

Any pointers would really help.

Kind regards

**Comment by swainn** [comment]
Hi @JamesWanjohi! The Home Page content is all managed through the Home Page Settings section of the Admin pages. Take a look at this documentation: http://docs.tethysplatform.org/en/stable/installation/production/configuration/advanced/customize.html#home-page-settings

This page also describes settings that can be used to change the theme of your Tethys Portal. Let me know if you have any specific questions after reviewing the documentation. 😄

**Reply by JamesWanjohi** [reply]
Hello @swainn!. Thank you for your prompt reply.

I mean I have a custom homepage that I have turned into a content page. This content is highly dynamic. 

1. I wish to persist this content to a database
2. The content is composed of texts and images
2. I wish to display this content dynamically on the web page

Is there a way to easily manage adding, removing and changing this content in Tethys, just like you would in a wordpress CMS?

**Comment by swainn** [comment]
@JamesWanjohi , unfortunately, the only wordpress-like features that Tethys has are those that I described above. Two options you may want to consider:

1. You could use the existing Home Page settings in Tethys to hold the content for your homepage and then use the variables in your custom home page template. Take a look at the source code for the Tethys home page to get an idea of how this would be done. The home page is composed of the following templates:
    * https://github.com/tethysplatform/tethys/blob/master/tethys_portal/templates/tethys_portal/home.html
    * https://github.com/tethysplatform/tethys/blob/master/tethys_portal/templates/header.html
    * https://github.com/tethysplatform/tethys/blob/master/tethys_portal/templates/footer.html
    * https://github.com/tethysplatform/tethys/blob/master/tethys_portal/templates/base.html

Anywhere you see the `site_globals` variable being used, it is referencing one of the values stored in the Home Page/Site Settings settings. For example:

```django
<div class="feature one">
  <img src="{% static site_globals.feature_1_image|default:'tethys_portal/images/placeholder.gif' %}" alt="">
  <h4>{{ site_globals.feature_1_heading|default:"Feature 1" }}</h4>
  <span>{{ site_globals.feature_1_body|default:"Use these features to brag about all of the things users can do with your instance of Tethys Portal." }}</span>
</div>
```

2. Alternatively, you could develop a custom Django app (it's like an extension for Django) that adds your home page and custom database models for storing the content. Take a look at the Django app tutorial here, for an idea of how to get started: https://docs.djangoproject.com/en/2.2/intro/tutorial01/

To add your Django app to Tethys, just install it like a normal Python package and then add it to the `INSTALLED_APPS` setting in your `portal_config.yml`.
